src/transaction_builder.js

```diff
@@ -153,33 +172,284 @@ export class TransactionBuilder {
+   * {TransactionBuilder}
+   * @see TransactionPreconditions
+   */
+  setMinAccountSequence(minAccountSequence) {
+    if (this.minAccountSequence !== null) {
+      throw new Error(
+        'min_account_sequence has been already set - setting min_account_sequence would overwrite it.'
+      );
+    }
+
+    this.minAccountSequence = minAccountSequence;
+
+    return this;
+  }
+
+  /**
+   * For the transaction to be valid, the current ledger time must
+   * be at least minAccountSequenceAge greater than sourceAccount's sequenceTime.
+   * Internally this will set the <code>minAccountSequenceAge</code> precondition.
+   *
+   * @param {number} durationInSeconds The minimum amount of time between
+   * source account sequence time and the ledger time when this transaction
+   * will become valid. If the value is `0`, the transaction is unrestricted by
+   * the account sequence age. Cannot be negative.
+   * @return {TransactionBuilder}
+   * @see TransactionPreconditions
+   */
+  setMinAccountSequenceAge(durationInSeconds) {
+    if (this.minAccountSequenceAge !== null) {
+      throw new Error(
+        'min_account_sequence_age has been already set - setting min_account_sequence_age would overwrite it.'
+      );
+    }
+
+    if (durationInSeconds < 0) {
+      throw new Error('min_account_sequence_age cannot be negative');
+    }
+
+    this.minAccountSequenceAge = durationInSeconds;
+
+    return this;
+  }
+
+  /**
+   * For the transaction to be valid, the current ledger number must be at
+   * least minAccountSequenceLedgerGap greater than sourceAccount's ledger
+   * sequence.
+   * Internally this will set the <code>minAccountSequenceLedgerGap</code>
+   * precondition.
+   *
+   * @param {number} gap The minimum number of ledgers between source account sequence and the ledger number when this transaction will become valid.
+   * If the value is `0`, the transaction is unrestricted by the account sequence ledger. Cannot be negative.
+   * @return {TransactionBuilder}
+   * @see TransactionPreconditions
+   */
+  setMinAccountSequenceLedgerGap(gap) {
+    if (this.minAccountSequenceLedgerGap !== null) {
+      throw new Error(
+        'min_account_sequence_ledger_gap has been already set - setting min_account_sequence_ledger_gap would overwrite it.'
+      );
+    }
+
+    if (gap < 0) {
+      throw new Error('min_account_sequence_ledger_gap cannot be negative');
+    }
+
+    this.minAccountSequenceLedgerGap = gap;
+
+    return this;
+  }
+
+  /**
+   * For the transaction to be valid, there must be a signature corresponding
+   * to every Signer in this array, even if the signature is not otherwise
+   * required by the sourceAccount or operations.
+   * Internally this will set the <code>extraSigners</code> precondition.
+   *
+   * @param {string[]} extraSigners The required extra signers.
+   * @return {TransactionBuilder}
+   * @see TransactionPreconditions
+   */
+  setExtraSigners(extraSigners) {
```

Comment:
NIT: it may be worth checking that `extraSigners` is an array and throwing an error if not. If it's just a string, for example, it will throw an error on LN 540 instead of defaulting to `[]`